src/main/java/com/revature/services/EvalLogicImpl.java

```diff
@@ -16,66 +24,260 @@
+        return evals;
 	}
 	
-//UPDATE-------------------------------
+	@Override
+	public EvalComment getCommentById(Integer id) {
+
+        EvalComment comment = commentDao.findOne(id);
+
+        if(comment == null){
+	        throw new NotFoundException("Comment with id " + id + " not found");
+		}
+
+		return comment;
+	}
+	
+//UPDATE-------------------------------	
+	@Override
+	@Transactional
+	public Eval updateEval(Eval eval, Integer id) {
+		Eval currEval = getEvalById(id);
+		
+		if(eval.getBatch() != null){
+            validation.validateBatchExists(eval.getBatch().getId());
+			currEval.setBatch(eval.getBatch());
+		}
+		if(eval.getDate() != null){
+			currEval.setDate(eval.getDate());
+		}
+		if(eval.getEvalType() != null){
+            validation.validateEvalTypeExists(eval.getEvalType().getId());
+			currEval.setEvalType(eval.getEvalType());
+		}
+		if(eval.getTrainee() != null){
+            validation.validateTraineeExists(eval.getTrainee().getId());
+			currEval.setTrainee(eval.getTrainee());
+		}
+		if(eval.getWeek() != null){
+            validation.validateWeek(eval.getWeek());
+			currEval.setWeek(eval.getWeek());
+		}
+        
+        dao.saveAndFlush(currEval);
+        entityManager.refresh(currEval);
+        return currEval;
+	}
+	
+	@Override
+	@Transactional
+	public EvalComment updateComment(EvalComment comment, Integer id) {
+		
+		EvalComment currComment = getCommentById(id);
+		
+		if(comment.getCommentText() != null){
+			currComment.setCommentText(comment.getCommentText());
+		}
+		
+		return commentDao.save(currComment);
+	}
 	
 //DELETE-------------------------------
+	@Override
+	@Transactional
+	public String deleteEval(Integer id) {
+		Eval eval = getEvalById(id);
+
+		dao.delete(eval);
+		return "Evaluation: " + id + " - DELETED";
+	}
+	
+	@Override
+	@Transactional
+	public String deleteComment(Integer id) {
+		EvalComment comment = getCommentById(id);
+				
+		commentDao.delete(comment);
+		return "Evaluation Comment: " + id + " - DELETED";
+	}
+
+    // private utility methods
+    
+    private void calculateTotalScores(Eval eval) {
+        int communicationScore = 0;
+        int knowledgeScore = 0;
+        int maxCommunicationScore = 0;
+        int maxKnowledgeScore = 0;
+        for (QuestionEval question : eval.getQuestions()) {
+            communicationScore += question.getCommunicationScore();
+            knowledgeScore += question.getKnowledgeScore();
+            maxCommunicationScore += question.getQuestionPool().getMaxCommunicationScore();
+            maxKnowledgeScore += question.getQuestionPool().getMaxKnowledgeScore();
+        }
+
+        eval.setEvalCommunicationScore(communicationScore);
+        eval.setEvalKnowledgeScore(knowledgeScore);
+        eval.setEvalMaxCommunicationScore(maxCommunicationScore);
+        eval.setEvalMaxKnowledgeScore(maxKnowledgeScore);
+    }
+
+    private void calculateAllTotalScores(Page<Eval> evals) {
+        evals.forEach((eval) -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1602)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "eval" parameter [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1611)